paramiko/client.py

```diff
@@ -279,6 +282,12 @@ def connect(
             The targets name in the kerberos database. default: hostname
         :param float banner_timeout: an optional timeout (in seconds) to wait
             for the SSH banner to be presented.
+        :param str pkcs11pin: If using PKCS11, this will be the pin of your
+            token or smartcard.
+        :param str pkcs11provider: If using PKCS11, this will be the provider
+            for the PKCS11 interface. Example: /usr/local/lib/opensc-pkcs11.so.
+        :param str pkcs11session: If using PKCS11 in a multithreaded
+            application you can share the session between threads.
```

Comment:
Can this be reduced to just one additional argument to `SSHClient.connect()`? Perhaps the user could be required to call `pkcs11_open_session()` themselves, then pass just the session to `connect()`?